worker/upload.go

```diff
@@ -444,6 +446,40 @@ loop:
 	return nil
 }
 
+func (mgr *uploadManager) renewUploader(u *uploader) {
+	// fetch renewed contract
+	fcid, _, _ := u.contractInfo()
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	renewed, err := mgr.b.RenewedContract(ctx, fcid)
+	cancel()
+
+	// remove the uploader if we can't renew it
+	mgr.mu.Lock()
+	if err != nil {
+		mgr.logger.Errorf("failed to fetch renewed contract for uploader %v: %v", fcid, err)
+		for i := 0; i < len(mgr.uploaders); i++ {
+			if mgr.uploaders[i] == u {
+				mgr.uploaders = append(mgr.uploaders[:i], mgr.uploaders[i+1:]...)
+				go u.Stop()
```

Comment:
I don't think there is a reason for this happening in a goroutine?

Reply:
Hm well the reason was that we `fail` every job in the queue was may or may not block a bit on sending the error response down the channel. This should be extremely rare so I figured instead of risking this to be a blocking thing, just `stop` in a goroutine would be fine. Removed it though.